linera-chain/src/manager.rs

```diff
@@ -385,36 +389,46 @@ impl ChainManager {
         Ok(Outcome::Accept)
     }
 
+    /// If the validated block certificate is more recent, return that certificate
+    /// so that the locked block can be updated.
+    pub fn maybe_update_locked(
+        &self,
+        validated_block_certificate: Option<LiteCertificate<'static>>,
+        executed_block: ExecutedBlock,
+    ) -> Option<Certificate> {
+        if let Some(lite_cert) = validated_block_certificate {
+            if self
+                .locked
+                .as_ref()
+                .map_or(true, |locked| locked.round < lite_cert.round)
+            {
+                let value = HashedCertificateValue::new_validated(executed_block);
+                return lite_cert.with_value(value);
+            }
+        }
+
+        None
+    }
+
     /// Signs a vote to validate the proposed block.
     pub fn create_vote(
         &mut self,
         proposal: BlockProposal,
-        outcome: BlockExecutionOutcome,
+        executed_block: ExecutedBlock,
         key_pair: Option<&KeyPair>,
         local_time: Timestamp,
+        used_blobs: Vec<Blob>,
+        maybe_update_locked: Option<Certificate>,
     ) {
+        let proposal_content = proposal.content.clone();
+        let published_blobs = proposal.published_blobs.clone();
         // Record the proposed block, so it can be supplied to clients that request it.
-        self.proposed = Some(proposal.clone());
+        self.update_proposed(proposal);
         self.update_current_round(local_time);
-        let ProposalContent { block, round, .. } = proposal.content;
-        let executed_block = outcome.with(block);
-
-        // If the validated block certificate is more recent, update our locked block.
-        if let Some(lite_cert) = proposal.validated_block_certificate {
-            if self
-                .locked
-                .as_ref()
-                .map_or(true, |locked| locked.round < lite_cert.round)
-            {
-                let value = HashedCertificateValue::new_validated(executed_block.clone());
-                if let Some(certificate) = lite_cert.with_value(value) {
-                    self.locked = Some(certificate.into());
-                }
-            }
-        }
+        let ProposalContent { round, .. } = proposal_content;
```

Comment:
Now instead of building the `ExecutedBlock` here we do it outside the function. This was done to make the whole `maybe_update_locked` work and not pass `used_blobs` that won't be used here. Maybe there's a better way? This was the best I could think of.
I could also pass in the round, the content and the blobs as separate arguments, instead of the whole proposal